File: Problems/215/215_Kth_Largest_Element_in_an_Array.py
from typing import List
from heapq import heappop, heappush
import heapq
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        klargest = [float('-Inf')]*k
        heapq.heapify(klargest)
        for num in nums:
            heapq.heappushpop(klargest, num)
        return klargest[0]

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        h = []
        for num in nums:
            heappush(h, num)
            if len(h) > k:
                heappop(h)
        return heappop(h)

sol = Solution()
nums = [3,2,3,1,2,4,5,5,6]
k = 4
print(sol.findKthLargest(nums, k))

File: Problems/215/215_Kth_Largest_Element_in_an_Array.cpp
#include<vector>
#include<iostream>
#include<queue>

using namespace std;

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        for(int i=0; i<k; i++){
            pq.push(INT_MIN);
        }
        for(auto num: nums){
            pq.push(num);
            pq.pop();
        }
        return pq.top();
    }
};